### ISSUE TITLE
No signature of method: java.lang.String.uncapitalize()

### ISSUE THREAD

**Issue by sdelamo** [energizedwork/webdriver-binaries-gradle-plugin#4]
When trying to use the Gradle Plugin in a Grails 3.3.1 app I am getting the next error: 

```
A problem occurred evaluating project ':complete'.
> Failed to apply plugin [id 'com.energizedwork.webdriver-binaries']
   > No signature of method: java.lang.String.uncapitalize() is applicable for argument types: () values: []
     Possible solutions: capitalize()
```

Uncapitalize seems to be included in Groovy since 2.4.8. Grails 3.3.1 uses Groovy 2.4.11 and Gradle 3.5. However, I am getting this error in IntelliJ IDEA 2017.2.6 Ultimate.

**Comment by erdi** [comment]
Are you getting the same error when running the build using gradle on the
command line?

**Comment by sdelamo** [comment]
No, I don't get it in command line.

**Comment by erdi** [comment]
This tells me that it's a problem with your setup and not with the plugin itself. Are you using the right Gradle version in IntelliJ?

**Comment by erdi** [comment]
You should probably have set it up to use the wrapper which you have as part of the repo.

**Comment by sdelamo** [comment]
Yes, I am using the wrapper. 

![preferences_and_build_gradle_-_complete_-____git_grailsguides_grails-geb-multiple-browsers_complete_](https://user-images.githubusercontent.com/864788/33063766-bd9a179e-cea3-11e7-91c7-7db3780e32c0.png)

**Comment by erdi** [comment]
I see. I would still prefer to understand why the version of Groovy used when you are running your build from IntelliJ is not the one that ships with Gradle 3.5 instead of just trying not to use certain methods. If we were to go down the route you've taken in your PR,  I'd also like to have an integration spec guarding not using `uncapitalize()` because the reasoning behind it will be lost quite quickly and somebody might question the point of that utils class and remove it.

**Comment by erdi** [comment]
Were you able to figure out why exactly the plugin is failing for you when using it through IntelliJ, @sdelamo?

**Comment by sdelamo** [comment]
Last friday we release Grails 3.3.2. It includes `webdriver-binaries-gradle-plugin` in the web profile by default. Some users have got this issue as I feared. It seems an issue with SDK pointing to a lower version of Gradle. 

See: https://github.com/grails/grails-core/issues/10877

I would prefer to avoid the use of uncapitalize() to improve a better user experience but I believe we can close this issue.

**Comment by erdi** [comment]
While I agree that it's a bit silly that people get failures stemming from the plugin code I would argue that it's not the plugin that is to blame. If people had a 2.x or 4.x version of Gradle on their path something else would probably fail. I therefore agree with [Eric Helgeson's comment](https://github.com/grails/grails-core/issues/10877#issuecomment-347248003) that a better behaviour with regards to which version of Gradle is used by default would probably be a good idea.

Anyhow, I'm happy to release a new, Gradle 3.0 compatible version of the plugin which could be used in the web profile for the next release of Grails. Hopefully this will help to prevent others hitting similar issues in the future.

**Comment by erdi** [comment]
Ping, @sdelamo

**Comment by sdelamo** [comment]
It would be great to have a Gradle 3.0 compatible version. I don't know how much work it would require on your side. Grails 3.2.x branch ships with Gradle 3.4.1 so we could add a `webdriver-binaries` dependencies to the default Grails 3.2.x web profile

**Comment by erdi** [comment]
Hopefully it won't be too much work :). I'll look into making it 3.0
compatible.

**Comment by erdi** [comment]
I've published [version 1.2](https://plugins.gradle.org/plugin/com.energizedwork.webdriver-binaries/1.2) which is Gradle 3.0 compatible.
